src/Mod/Assembly/CommandCreateSimulation.py

```diff
@@ -0,0 +1,1019 @@
+        return None
+
+
+class ViewProviderMotion:
+    def __init__(self, vp):
+        vp.Proxy = self
+        self.updateLabel()
+
+    def attach(self, vpDoc):
+        """Setup the scene sub-graph of the view provider, this method is mandatory"""
+        self.app_obj = vpDoc.Object
+        self.assembly = self.app_obj.Proxy.getAssembly(self.app_obj)
+
+        self.display_mode = coin.SoType.fromName("SoFCSelection").createInstance()
+
+        vpDoc.addDisplayMode(self.display_mode, "Wireframe")
+
+    def updateData(self, feaPy, prop):
+        """If a property of the handled feature has changed we have the chance to handle this here"""
+        pass
+
+    def getDisplayModes(self, vpDoc):
+        """Return a list of display modes."""
+        return ["Wireframe"]
+
+    def getDefaultDisplayMode(self):
+        """Return the name of the default display mode. It must be defined in getDisplayModes."""
+        return "Wireframe"
+
+    def onChanged(self, vpDoc, prop):
+        """Here we can do something when a single property got changed"""
+        # App.Console.PrintMessage("Change property: " + str(prop) + "\n")
+        pass
+
+    def getIcon(self):
+        if self.app_obj.MotionType == "Angular":
+            return ":/icons/button_rotate.svg"
+
+        return ":/icons/button_right.svg"
+
+    def dumps(self):
+        """When saving the document this object gets stored using Python's json module.\
+                Since we have some un-serializable parts here -- the Coin stuff -- we must define this method\
+                to return a tuple of all serializable objects or None."""
+        return None
+
+    def loads(self, state):
+        """When restoring the serialized object from document we have the chance to set some internals here.\
+                Since no data were serialized nothing needs to be done here."""
+        return None
+
+    def doubleClicked(self, vpDoc):
+        if self.assembly is None:
+            return False
+
+        if UtilsAssembly.activeAssembly() != self.assembly:
+            Gui.ActiveDocument.setEdit(self.assembly)
+
+        self.openEditDialog()
+
+    def openEditDialog(self):
+        joint = None
+        if self.app_obj.Joint is not None:
+            joint = self.app_obj.Joint[0]
+
+        dialog = MotionEditDialog(
+            self.assembly, self.app_obj.MotionType, joint, self.app_obj.Formula
+        )
+        if dialog.exec_():
+            self.app_obj.MotionType = dialog.motionType
+            self.app_obj.Joint = dialog.joint
+            self.app_obj.Formula = dialog.formula
+
+            self.updateLabel()
+
+    def updateLabel(self):
+        if self.app_obj.Joint is None:
+            return
+
+        label = self.app_obj.Joint[0].Label + "("
+        if self.app_obj.MotionType == "Linear":
+            label = label + translate("Assembly", "Linear")
+        else:
+            label = label + translate("Assembly", "Angular")
+        label = label + ")"
```

Comment:
Couldn't it be done like:
```
label = f"({translate("Assembly", "Linear" if self.app_obj.MotionType == "Linear" else "Angular")})"
```

?

Reply:
Your formula is missing `self.app_obj.Joint[0].Label +` which would make the line super long so auto format will split this in several lines which imo will end up being less readable.

Reply:
What about:
```python
        type = "Linear" if self.app_obj.MotionType == "Linear" else "Angular"

        self.app_obj.Label = "{label} ({type})".format(
            label=self.app_obj.Joint[0].Label,
            type=translate("Assembly", type)
        )
```

Seems more readable and it is easier to understand how the expected format looks like

Reply:
Can translate take a variable? I seem to recall that some translate function can not. But maybe it's one of the cpp

Reply:
translate as function should not have a problem with that - but tool that extracts strings to translate can. If this strings are not used anywhere else they should be probably marked for translation somewhere else with no-op translation. Not sure in this case - good call.

Reply:
Ah well I have included your patch now so hopefully it's ok :)

Reply:
We'll see! If there will be a problem, I'll add the required QT_TRANSLATE_NOOPs :)